lib/init/config-file.js

```diff
@@ -83,7 +83,7 @@ async function writeYAMLConfigFile(config, filePath) {
 async function writeJSConfigFile(config, filePath) {
     debug(`Writing JS config file: ${filePath}`);
 
-    const stringifiedContent = `module.exports = ${stringify(config, { cmp: sortByKey, space: 4 })}\n`;
+    const stringifiedContent = `/* globals module */\nmodule.exports = ${stringify(config, { cmp: sortByKey, space: 4 })}\n`;
```

Comment:
I'm having a hard time understanding what you're saying here. 

The error message is always output for JS config files, I don't think the point was to eliminate that message. The point was to eliminate a linting error.

Are you saying something else is already declaring `module`?

Reply:
yes, it's happening when
* `globals: {module: true}` or `env: { commonjs: true}`
* rule `no-redeclare` was enabled

In this case, it will introduce a new linting error.

Reply:
I can see a few options to resolve it:

1. adding `/*eslint env: node*/`, the drawback is that `env` will go away in the new config.
2. adding `/* eslint-disable-next-line no-undef`*/.  the comment will be removed by the fixer if the gloabal `module` has been declared in the eslintrc, e.g. `eslint-config-airbnb`.
3. adding a config item for the `.eslintrc.js`, something like 
```js
module.exports = {
  ...,
  overrides: [{files: [".eslintrc.cjs"], globals: {"module": true}}],
};
```

Reply:
Solution 3 with `overrides` sounds good to me.